Netstat.pm

```diff
@@ -0,0 +1,243 @@
+# put common statistics parsing functions in a module
+
+# Copyright (c) 2023 Moritz Buhl <mbuhl@genua.de>
+#
+# Permission to use, copy, modify, and distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+package Netstat;
+
+use strict;
+use warnings;
+
+use parent 'Exporter';
+our @EXPORT= qw(
+    generate_diff_netstat_s
+);
+
+# netstat output might use plural for some nouns always use
+# the plural form of these words.
+sub canonicalize_key {
+    my ($key) = @_;
+    my @plurales = qw(miss);
+    my @pluralys = qw(entr);
+    my @plurals = qw(ACK Interface SYN TDB accept ack agreement allocation
+	association attempt authentication byte calculation call change
+	checksum cleanup collision connection datagram decapsulation
+	decryption destination drop duplicate episode error failure field
+	flow fragment frame gap gateway global insert jump llx local
+	lookup mbuf message mismatche node notification option overflow
+	packet prediction probe quer redirect replay report request
+	response rexmit route scan seed segment slide state table
+	timeout transition upcall use);
+
+    $key =~ s/miss/misses/;
+    $key =~ s/entry/entries/;
+    foreach my $sub (@plurals) {
+	$key =~ s/ ${sub}([^s])/ ${sub}$1/;
+	$key =~ s/^${sub}([^s])/${sub}s$1/;
+    }
+
+    chomp $key;
+    return $key;
+}
+
+# map netstat indentation to perl hash some nestat -s lines increase the
+# indentation and at the same time have a value.  Write that value to the
+# "total" field.
+sub parse {
+    my ($fname) = @_;
+    my %netstat;
+    my $l1;
+    my $l2;
+    my $l3;
+    open(my $fh, '<', "$fname") or die $!;
```

Comment:
Useless use of quotes: "$fname"
Please add a description to die().  If it fails looking up the line in the source makes it harder to debug.
I usually use: or die "Open '$fname' for reading failed: $!";